src/central-server/admin-service/core/src/main/java/org/niis/xroad/cs/admin/core/service/NotificationServiceImpl.java

```diff
@@ -0,0 +1,139 @@
+
+@Service
+@RequiredArgsConstructor
+public class NotificationServiceImpl implements NotificationService {
+
+    private final ConfigurationSigningKeysService configurationSigningKeysService;
+    private final SystemParameterService systemParameterService;
+    private final SignerProxyFacade signerProxyFacade;
+    private final GlobalConfGenerationStatusService globalConfGenerationStatus;
+
+    @Override
+    public Set<AlertInfo> getAlerts() {
+
+        final List<TokenInfo> tokens;
+        try {
+            tokens = signerProxyFacade.getTokens();
+        } catch (Exception e) {
+            return Set.of(new AlertInfo("status.signer_error"));
+        }
+
+        final Set<AlertInfo> alerts = new HashSet<>();
+        if (isInitialized(tokens)) {
+            alerts.addAll(checkGlobalConfGenerationStatus());
+            alerts.addAll(checkConfigurationSigningKey(SOURCE_TYPE_INTERNAL, tokens));
+            if (getCenterTrustedAnchorsAllowed()) {
+                alerts.addAll(checkConfigurationSigningKey(SOURCE_TYPE_EXTERNAL, tokens));
+            }
+        }
+        return alerts;
+    }
+
+    private boolean isInitialized(List<TokenInfo> tokens) {
+        return isNotBlank(systemParameterService.getInstanceIdentifier())
+                && isNotBlank(systemParameterService.getCentralServerAddress())
+                && softwareTokenInitialized(tokens);
+    }
+
+    private boolean softwareTokenInitialized(List<TokenInfo> tokens) {
+        return tokens.stream()
+                .anyMatch(token -> token.getId().equals(SOFTWARE_TOKEN_ID) && token.getStatus() != NOT_INITIALIZED);
+    }
+
+    private Set<AlertInfo> checkGlobalConfGenerationStatus() {
+        final Optional<GlobalConfGenerationStatus> confStatus = globalConfGenerationStatus.get();
+
+        if (confStatus.isEmpty()) {
+            return Set.of(new AlertInfo("status.global_conf_generation.status_not_found"));
+        }
+
+        final GlobalConfGenerationStatus status = confStatus.get();
+        if (!status.isSuccess()) {
+            return Set.of(new AlertInfo("status.global_conf_generation.failing", status.getTime()));
+        }
+        if (SECONDS.between(status.getTime(), Instant.now()) > systemParameterService.getConfExpireIntervalSeconds()) {
+            return Set.of(new AlertInfo("status.global_conf_generation.global_conf_expired"));
+        }
+
+        return Set.of();
+    }
+
+    private Set<AlertInfo> checkConfigurationSigningKey(String sourceType, List<TokenInfo> tokens) {
+        final Optional<ConfigurationSigningKey> signingKeyOptional = configurationSigningKeysService.findActiveForSource(sourceType);
+        if (signingKeyOptional.isEmpty()) {
+            return Set.of(new AlertInfo(format("status.signing_key.%s.missing", sourceType)));
+        }
+        final ConfigurationSigningKey signingKey = signingKeyOptional.get();
+        for (final TokenInfo token : tokens) {
```

Comment:
did this pass sonarqube complexity check? Might be hard to simplify, but worth considering.